dissect/target/plugins/os/windows/log/mssql.py

```diff
@@ -81,19 +81,38 @@ def errorlog(self) -> Iterator[MssqlErrorlogRecord]:
                 buf = ""
 
                 for line in fh:
-                    if ts := RE_TIMESTAMP_PATTERN.match(line):
+                    # If we have a buffer with a timestamp and
+                    # our current line also has a timestamp,
+                    # we should have a complete record in our buffer.
+                    if previous_ts := RE_TIMESTAMP_PATTERN.match(buf):
+                        if current_ts := RE_TIMESTAMP_PATTERN.match(line):
+                            yield MssqlErrorlogRecord(
+                                ts=datetime.strptime(previous_ts.group(), "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=timezone.utc),
+                                instance=instance,
+                                # The process name is a fixed-width field and is always 12 characters long.
+                                process=buf[23:35].strip(),
+                                message=buf[35:].strip(),
+                                path=errorlog,
+                                _target=self.target,
+                            )
+
+                            buf = ""
+
+                    buf += line
+
+                # For the last line
+                if buf:
+                    if current_ts := RE_TIMESTAMP_PATTERN.match(line):
```

Comment:
This only works if the last line is timestamped. If it is text, it will fail and the last record will be dropped.

The solution is to use `current_ts`